### PR TITLE
(doc) Add Puppet Server 2.1.2 release notes.

### PR DESCRIPTION
**Pull request by gguillotte**
Add Puppet Server 2.1.2 release notes and make some style and formatting edits to older notes. @camlow325

### CODE REVIEW

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -10,11 +10,49 @@ canonical: "/puppetserver/latest/release_notes.html"
...
+[deprecated]: ./deprecated_features.html
+[Puppet `auth.conf` file]: /puppet/latest/reference/config_file_auth.html
+[`auth.conf`]: ./config_file_auth.html
+[`ca.conf`]: ./config_file_ca.html
+[`master.conf`]: ./config_file_master.html
 
+## Puppet Server 2.1.2
+
+Released October 19, 2015.
+
```
Comment (camlow325) [inline documentation/release_notes.markdown]: @gguillotte Should we have a supported platforms section in here?  Should be the same as with the last release other than `Debian 8` being added.

**Comment by camlow325** [comment]
:+1: other than the question about the supported platforms section.  @dankreek - WDYT?

**Comment by dankreek** [comment]
:+1: Looks good!

**Comment by gguillotte** [comment]
Added Debian 8.

**Comment by camlow325** [comment]
:+1:

**Comment by camlow325** [comment]
Packages are live now, so merging...

**Pull request merged**